src/librustc_interface/util.rs

```diff
@@ -42,12 +42,15 @@ use std::{panic, thread};
 /// features is available on the target machine, by querying LLVM.
 pub fn add_configuration(
     cfg: &mut CrateConfig,
-    sess: &Session,
+    sess: &mut Session,
     codegen_backend: &dyn CodegenBackend,
 ) {
     let tf = sym::target_feature;
 
-    cfg.extend(codegen_backend.target_features(sess).into_iter().map(|feat| (tf, Some(feat))));
+    let target_features = codegen_backend.target_features(sess);
+    sess.target_features.extend(target_features.iter().cloned());
```

Comment:
Can you read this from `sess.parse_sess.config` instead where `sess.target_features` is read right now?

Reply:
I am hesitant to rely on that since you can override it on the command-line with `--cfg target_arch="foo"`.